app/redux/wallet/reducer.js

```diff
@@ -84,7 +68,19 @@ const reducer = (state: StoreStateType = initialState, action: Action) => {
     }
     case SET_TRANSACTIONS: {
       const { transactions } = action.payload;
-      return { ...state, transactions };
+      const getFirstUniqueTransactions = (txList: TxList, n: number = 3): TxList => {
```

Comment:
don't mix updating different store values in one reducer

Reply:
Will be separated to a different reducer and action.
Do you mean saved contact from last used or Tx right? If we add a contact from contacts page (not from last used) it shouldn't be added to last used, correct?

Reply:
> If we add a contact from contacts page (not from last used) it shouldn't be added to last used, correct?

Correct, the only case when adding to latest used addresses is actually sending tx. But in case of saving new contact, latest addresses have to be checked and updated if necessary

Reply:
np